src/main/java/Duke.java

```diff
@@ -1,10 +1,35 @@
+import duke.*;
+
+import java.io.File;
+import java.io.IOException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String path, String fileName) {
+        this.storage = new Storage(path,fileName);
+        this.tasks = storage.load();
+        this.ui = new Ui(this.tasks);
+    }
+
+    public void run() throws IOException, EmptyInputException, NoResponseException {
+        storage.createFie();
+        ui.showWelcome();
+        Parser parser = new Parser(this.ui,this.tasks);
+        while (!parser.isEnd) {
+            String fullCommand = ui.readCommand();
+            parser.parse(fullCommand);
+        }
+        storage.storeFile(this.tasks);
```

Comment:
Should Storage#storeFile be called after every command to save the task message in case of the application crash?

Reply:
thank you and edited

Reply:
Oh, I see. Perhaps can consider rewriting the whole task information to the file after every add, edit and delete operation.